pkg/cedar/intake/client.go

```diff
@@ -106,49 +97,31 @@ func (c *Client) CheckConnection(ctx context.Context) error {
 	return nil
 }
 
-// PublishSnapshot sends the given current state of a SystemIntake and all its
-// associated entities to CEDAR for eventual storage in Alfabet
-func (c *Client) PublishSnapshot(
-	ctx context.Context,
-	si *models.SystemIntake,
-	bc *models.BusinessCase,
-	acts []*models.Action,
-	notes []*models.Note,
-	fbs []*models.GRTFeedback,
-) error {
-	id := uuid.Nil
-	if si != nil {
-		id = si.ID
-	}
+// PublishIntake sends an object to CEDAR through the Intake API for eventual storage in Alfabet
+func (c *Client) PublishIntake(ctx context.Context, model translation.IntakeObject) error {
```

Comment:
Something I thought of last night:

Right now, callers of this method would need to convert the system intake/business case/etc. object to the `Translatable*` version themselves; calling code would look something like
```go
systemIntake := // get a models.SystemIntake somehow

err := c.PublishIntake(ctx, translation.TranslatableSystemIntake(systemIntake))
```

This isn't horrible, but it does mean the `Translatable*` types "leak" out into the broader codebase.

What we could do instead is have utility methods on `Client` for each type of `IntakeObject` that convert to the appropriate `Translatable` type themselves, something like
```
func (c *Client) PublishSystemIntake(ctx context.Context, systemIntake models.SystemIntake) error {
  return c.publishIntake(ctx, translation.TranslatableSystemIntake(systemIntake))
}
```

which keeps the broader codebase from needing to know about the `Translatable` types.

Thoughts?

Reply:
I like this approach. Having `publishIntake` be private, and scoping the public methods to known types feels a lot cleaner.

Reply:
Done in https://github.com/CMSgov/easi-app/pull/1450/commits/4b5e39c245fe25911721d24b5a51bdc8007e2518.